pkg/qbittorrent/client.go

```diff
@@ -242,3 +263,50 @@ func (c *Client) setCookies(cookies []*http.Cookie) {
 	cookieURL, _ := url.Parse(fmt.Sprintf("%v://%v:%v", c.settings.protocol, c.settings.Hostname, c.settings.Port))
 	c.http.Jar.SetCookies(cookieURL, cookies)
 }
+
+func buildUrl(settings Settings, endpoint string) string {
+	// parse url
+	u, _ := url.Parse(settings.Hostname)
+
+	// reset Opaque
+	u.Opaque = ""
+
+	// set scheme
+	scheme := "http"
```

Comment:
What's going on here?

Reply:
It's all unnecessarily complex because it needs to support old setups with port in the separate field, as well as only using the `host`. There's some testcases that cover most of it and with one of them like `qbit.domain.ltd` it would parse it incorrectly and set `u.Opaque` to http or qbit which was a bit weird.

The background of the change is I got some reports of it being broken when using qbit behind the reverse proxy Traefik.

Reply:
I had an issue I think with sonarr not working on port 80. Unrelated to this issue, but still something

Reply:
The branch below doesn't make sense because it's the same code